cpp/dolfinx/fem/Function.h

```diff
@@ -397,29 +392,22 @@ class Function
 
     // Interpolate point values on each cell (using last computed value if
     // not continuous, e.g. discontinuous Galerkin methods)
-    Eigen::Array<double, Eigen::Dynamic, 3, Eigen::RowMajor> x(num_dofs_g, 3);
-    Eigen::Array<T, Eigen::Dynamic, Eigen::Dynamic, Eigen::RowMajor> values(
-        num_dofs_g, value_size_loc);
     auto map = mesh->topology().index_map(tdim);
     assert(map);
     const std::int32_t num_cells = map->size_local() + map->num_ghosts();
+
+    Eigen::Array<int, Eigen::Dynamic, 1> cells(x_g.rows());
+
     for (std::int32_t c = 0; c < num_cells; ++c)
     {
       // Get coordinates for all points in cell
       auto dofs = x_dofmap.links(c);
       for (int i = 0; i < num_dofs_g; ++i)
-        x.row(i) = x_g.row(dofs[i]);
-
-      // Call evaluate function
-      Eigen::Array<int, Eigen::Dynamic, 1> cells(x.rows());
-      cells = c;
-      eval(x, cells, values);
-
-      // Copy values to array of point values
-      for (int i = 0; i < values.rows(); ++i)
-        point_values.row(dofs[i]) = values.row(i);
+        cells[dofs[i]] = c;
     }
 
+    eval(x_g, cells, point_values);
```

Comment:
If necessary, `Function::eval` interface for `cells` can be changed to `tcb::span`.